### PR TITLE
Show syncChangeCounter/syncStatus in client bookmark and add a SQL query runner tab.

### PR DESCRIPTION
**Pull request by thomcc**
As suggested yesterday.

### CODE REVIEW

File: data/components.js [hunk 1]
```diff
@@ -4,6 +4,8 @@ Cu.import("resource://gre/modules/Services.jsm");
 Cu.import("resource://gre/modules/FxAccounts.jsm");
 Cu.import("resource://services-sync/main.js");
 Cu.import("resource://gre/modules/Task.jsm");
+Cu.import("resource://services-common/async.js");
```
Comment (mhammond) [inline data/components.js]: do we need this?

**Comment by mhammond** [comment]
(hopefully you can make that change and merge yourself? I'm not very good at githubbing!)

File: data/components.js (hunk 1 above)
Comment (thomcc) [inline data/components.js]: My bad, it's used in commit 3, but the order I did these commits isn't exactly reflected in the log.

**Comment by thomcc** [comment]
And I'm not sure the validator reporting syncStatus/syncChangeCounter would be that meaningful. If those having unexpected values would cause an error, it does report that. Otherwise ISTM like we don't really care.

File: data/components.js (hunk 1 above)
Comment (mhammond) [inline data/components.js]: > My bad, it's used in commit 3, but the order I did these commits isn't exactly reflected in the log.

oops, I missed that - but IIUC, it should be easy to have that function return a promise instead od spinning there - the caller is already a task.

**Comment by mhammond** [comment]
> And I'm not sure the validator reporting syncStatus/syncChangeCounter would be that meaningful. If those having unexpected values would cause an error, it does report that. Otherwise ISTM like we don't really care.

I guess I'm just saying that an invalid status or change counter implies there is a sync pending, so the validation results are likely to be inaccurate

**Comment by thomcc** [comment]
Fixed, but i'm closing this PR and resubmitting one since I can't find whatever button is "rerequest review"

**Pull request closed without merging**